core/changedetector/proto_change_detector_service.go

```diff
@@ -0,0 +1,215 @@
+		SchemaNames: impactedDependents,
+	}
+}
+
+func setDirectlyImpactedSchemasAndFields(currentFds, prevFds *descriptor.FileDescriptorSet, sce *stencilv1beta2.SchemaChangedEvent) {
+	packageMessageMap := getPackageMessageMap(prevFds)
+	packageEnumMap := getPackageEnumMap(prevFds)
+	for _, fds := range currentFds.GetFile() {
+		for _, newMessageDesc := range fds.GetMessageType() {
+			messageName := fds.GetPackage() + "." + newMessageDesc.GetName()
+			if oldMessageDesc := getMessageDescriptor(packageMessageMap, fds.GetPackage(), newMessageDesc.GetName()); oldMessageDesc != nil {
+				if !proto.Equal(newMessageDesc, oldMessageDesc) {
+					sce.UpdatedSchemas = append(sce.UpdatedSchemas, messageName)
+					appendImpactedFields(sce, messageName, getImpactedMessageFields(oldMessageDesc, newMessageDesc))
+				}
+				for _, newEnumDesc := range newMessageDesc.GetEnumType() {
+					enumName := messageName + "." + newEnumDesc.GetName()
+					if oldEnumDesc := findEnumDescriptorFromMessageDescriptor(oldMessageDesc, newEnumDesc.GetName()); oldEnumDesc != nil {
+						if !proto.Equal(newEnumDesc, oldEnumDesc) {
+							sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+							appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+						}
+					} else {
+						sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+						appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+					}
+				}
+			} else {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, messageName)
+				appendImpactedFields(sce, messageName, getImpactedMessageFields(oldMessageDesc, newMessageDesc))
+			}
+		}
+		compareEnumDescriptors(fds, packageEnumMap, sce)
+	}
+}
+func appendImpactedFields(sce *stencilv1beta2.SchemaChangedEvent, key string, impactedFields []string) {
+	if sce.UpdatedFields == nil {
+		sce.UpdatedFields = make(map[string]*stencilv1beta2.ImpactedFields)
+	}
+	if val, ok := sce.UpdatedFields[key]; ok {
+		val.FieldNames = append(val.FieldNames, impactedFields...)
+	} else {
+		sce.UpdatedFields[key] = &stencilv1beta2.ImpactedFields{
+			FieldNames: impactedFields,
+		}
+	}
+}
+
+func compareEnumDescriptors(fds *descriptorpb.FileDescriptorProto, packageEnumMap map[string]map[string]*descriptor.EnumDescriptorProto, sce *stencilv1beta2.SchemaChangedEvent) {
+	for _, newEnumDesc := range fds.GetEnumType() {
+		enumName := fds.GetPackage() + "." + newEnumDesc.GetName()
+		if oldEnumDesc := getEnumDescriptor(packageEnumMap, fds.GetPackage(), newEnumDesc.GetName()); oldEnumDesc != nil {
+			if !proto.Equal(newEnumDesc, oldEnumDesc) {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+				appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+			}
+		} else {
+			sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+			appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+		}
+	}
```

Comment:
in this case `continue` here

```suggestion
	for _, newEnumDesc := range fds.GetEnumType() {
		enumName := fds.GetPackage() + "." + newEnumDesc.GetName()
		if oldEnumDesc := getEnumDescriptor(packageEnumMap, fds.GetPackage(), newEnumDesc.GetName()); oldEnumDesc != nil {
			if !proto.Equal(newEnumDesc, oldEnumDesc) {
				sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
				appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
			}
		  continue	
		}
			sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
			appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
	}
```

Reply:
Same as above. else here adding more readablity.
Also, this also makes sense as we have more things to do inside if condition
Ref: https://stackoverflow.com/questions/963982/should-i-use-return-continue-statement-instead-of-if-else#:~:text=My%20personal%20approach%20of%20choosing%20one%20is%20that%20if%20the%20body%20of%20the%20if%20part%20is%20very%20short%20(3%20or%204%20lines%20maximum)%2C%20it%20makes%20sense%20to%20use%20the%20return/continue%20variant.%20If%20the%20body%20is%20long%2C%20it%27s%20harder%20to%20keep%20track%20of%20the%20control%20flow%20so%20I%20choose%20the%20else%20version.

Reply:
updated

Reply:
thanks, resolving it